test/testenv/verificationutils.go

```diff
@@ -559,39 +559,70 @@ func VerifyPVCsPerDeployment(deployment *Deployment, testenvInstance *TestEnv, d
 	}
 }
 
-// VerifyAppInstall verify that app of specific version is installed
-func VerifyAppInstall(deployment *Deployment, testenvInstance *TestEnv, ns string, podName string, apps map[string]string, versionCheck bool, statusCheck string) {
-	for appName, version := range apps {
-		status, versionInstalled, err := GetPodAppStatus(deployment, podName, ns, appName)
-		logf.Log.Info("App info returned for app", "App-name", appName, "status", status, "versionInstalled", versionInstalled, "error", err)
-		gomega.Expect(err).To(gomega.Succeed(), "Unable to get app status on pod ")
-		gomega.Expect(strings.ToLower(status)).Should(gomega.Equal(strings.ToLower(statusCheck)))
-		if versionCheck {
-			gomega.Expect(versionInstalled).Should(gomega.Equal(version))
+// VerifyAppInstalled verify that app of specific version is installed
+func VerifyAppInstalled(deployment *Deployment, testenvInstance *TestEnv, ns string, pods []string, apps []string, versionCheck bool, statusCheck string, checkupdated bool, clusterWideInstall bool) {
+	for _, podName := range pods {
+		if !strings.Contains(podName, "monitoring-console") {
+			for _, app := range apps {
+				appName := AppInfo[app]["App-name"]
+				status, versionInstalled, err := GetPodAppStatus(deployment, podName, ns, appName)
+				logf.Log.Info("App info returned for app", "App-name", appName, "status", status, "versionInstalled", versionInstalled, "error", err)
+				gomega.Expect(err).To(gomega.Succeed(), "Unable to get app status on pod ")
+				comparsion := strings.EqualFold(status, statusCheck)
+				//Check the app is installed on specific pods and un-installed on others for cluster-wide install
+				if clusterWideInstall {
+					if strings.Contains(podName, "-indexer-") || strings.Contains(podName, "-search-head-") || strings.Contains(podName, "-standalone-") {
+						gomega.Expect(comparsion).Should(gomega.Equal(true))
+					} else {
+						gomega.Expect(comparsion).Should(gomega.Equal(false))
+					}
+				} else {
```

Comment:
This does not account for a case when installation is `local` and pod is indexer or search head.

Reply:
For local the installation would only be on one of the pods in 'CM, LM, SHC, Standalone'. In which case i think it would be better to pass that pod with status as enabled and pass the other pods  with status check as disabled

Reply:
You can do a similar logic in `else` that way test writer would pass all pods and will be done.

Reply:
modified